rbx_binary/src/deserializer.rs

```diff
@@ -1123,6 +1120,86 @@ impl<R: Read> BinaryDeserializer<R> {
                     })
                 }
             },
+            Type::OptionalCFrame => match canonical_type {
+                VariantType::OptionalCFrame => {
+                    let referents = &type_info.referents;
+                    let mut rotations = Vec::with_capacity(referents.len());
+
+                    // Roblox writes a type marker for CFrame here that we don't
+                    // need to use.
+                    chunk.read_u8()?;
+
+                    for _ in 0..referents.len() {
+                        let id = chunk.read_u8()?;
+                        if id == 0 {
+                            rotations.push(Matrix3::new(
+                                Vector3::new(
+                                    chunk.read_le_f32()?,
+                                    chunk.read_le_f32()?,
+                                    chunk.read_le_f32()?,
+                                ),
+                                Vector3::new(
+                                    chunk.read_le_f32()?,
+                                    chunk.read_le_f32()?,
+                                    chunk.read_le_f32()?,
+                                ),
+                                Vector3::new(
+                                    chunk.read_le_f32()?,
+                                    chunk.read_le_f32()?,
+                                    chunk.read_le_f32()?,
+                                ),
+                            ));
+                        } else if let Some(basic_rotation) = cframe::from_basic_rotation_id(id) {
+                            rotations.push(basic_rotation);
+                        } else {
+                            return Err(InnerError::BadRotationId {
+                                type_name: type_info.type_name.clone(),
+                                prop_name,
+                                id,
+                            });
+                        }
+                    }
+
+                    let mut x = vec![0.0; referents.len()];
+                    let mut y = vec![0.0; referents.len()];
+                    let mut z = vec![0.0; referents.len()];
+
+                    chunk.read_interleaved_f32_array(&mut x)?;
+                    chunk.read_interleaved_f32_array(&mut y)?;
+                    chunk.read_interleaved_f32_array(&mut z)?;
+
+                    // Roblox writes a type marker for Bool here that we don't
+                    // need to use.
+                    chunk.read_u8()?;
```

Comment:
Okay, while the other type marker might make sense, this seems pretty weird. I think we should also assert on its value and return an error if it doesn't match `Bool`.